encoders/src/test/scala/sql/EncoderDerivationSpec.scala

```diff
@@ -398,3 +402,33 @@ class EncoderDerivationSpec extends munit.FunSuite with SparkSqlTesting:
       )
     )
   }
+
+  test("Custom encoder to a primitive type wrapped in an Option") {
+    // Defining some custom Serializer and Deserializer for Instant to LongType
+    given Serializer[Instant] with
+      def inputType: DataType = ObjectType(classOf[Instant])
+      def serialize(inputObject: Expression): Expression =
+        val long = Invoke(inputObject, "toEpochMilli", LongType, returnNullable = false)
+        summon[Serializer[Long]].serialize(long)
+
+    given Deserializer[Instant] with
+      def inputType: DataType = LongType
+      def deserialize(path: Expression): Expression =
+        val long = summon[Deserializer[Long]].deserialize(path)
+        StaticInvoke(
+          classOf[Instant],
+          ObjectType(classOf[Instant]),
+          "ofEpochMilli",
+          long :: Nil,
+          returnNullable = false
+        )
+
+    val encoder = summon[Encoder[ChkCustom]]
+    assertEquals(
+      encoder.schema,
+      StructType(Seq(StructField("u", LongType, true)))
```

Comment:
without the Serializer/Deserializer above it would be `TimestampType` which would be correct I think

Reply:
Yes but this test is showing that it does not work if we custom redefine to something else.

Reply:
OK - Now I get it - you just check if you can override the default encoder with your own
OK - and a scenario like that wasn't tested before - and Option is a bit tricky here (again)

Reply:
I agree, Option is the trickiest because of `nullable` 😆